code/game/machinery/doors/airlock.dm

```diff
@@ -122,6 +122,9 @@ About the new airlock wires panel:
 *		one wire for controlling door speed.  When active, dor closes at normal rate.  When cut, door does not close manually.  When pulsed, door attempts to close every tick.
 */
 
+/obj/machinery/door/airlock/CanAStarPass(obj/item/card/id/ID, to_dir, atom/movable/caller)
```

Comment:
Conceptually I think it would be better if instead of passing in an ID card, it passed in GetAccess() (so a lazylist of accesses). Access can be held by things other than cards, supplied arbitrarily, combined from several sources, etc.